src/sdk/arithmetic.hpp

```diff
@@ -0,0 +1,94 @@
+#ifndef realm_sdk_arithmetic_hpp
+#define realm_sdk_arithmetic_hpp
+
+#include "primitive.hpp"
+
+namespace realm::sdk {
+
+template <typename ArithmeticT>
+struct ArithmeticBase
+: private PrimitiveBase<Property<ArithmeticT>, ArithmeticT>
+{
+    using base = PrimitiveBase<Property<ArithmeticT>, ArithmeticT>;
+    using typename base::value_type;
+    using base::operator=;
+    using base::is_managed;
+    using base::value;
+
+    friend base;
+    template <typename Impl>
+    friend struct ObjectBase;
+
+    ArithmeticBase& operator+=(ArithmeticT rhs) {
```

Comment:
You probably want a const-reference here. This will currently increment and decrement the `SharedRealm` refcount unnecessarily.